packages/backend/server/src/fundamentals/helpers/url.ts

```diff
@@ -6,19 +8,38 @@ import { Config } from '../config';
 @Injectable()
 export class URLHelper {
   private readonly redirectAllowHosts: string[];
-  readonly origin = this.config.node.dev
-    ? 'http://localhost:8080'
-    : `${this.config.server.https ? 'https' : 'http'}://${this.config.server.host}${
-        this.config.server.host === 'localhost' ||
-        this.config.server.host === '0.0.0.0'
-          ? `:${this.config.server.port}`
-          : ''
-      }`;
 
-  readonly baseUrl = `${this.origin}${this.config.server.path}`;
-  readonly home = this.baseUrl;
+  readonly origin: string;
+  readonly baseUrl: string;
+  readonly home: string;
 
   constructor(private readonly config: Config) {
+    if (this.config.server.externalUrl) {
+      if (!this.verify(this.config.server.externalUrl)) {
+        throw new Error(
+          'Invalid `server.externalUrl` configured. It must be a valid url.'
+        );
+      }
+
+      const externalUrl = new URL(this.config.server.externalUrl);
+
+      this.origin = externalUrl.origin;
+      this.baseUrl =
+        externalUrl.origin + externalUrl.pathname.replace(/\/$/, '');
+      console.log(externalUrl);
+    } else {
+      this.origin = [
+        this.config.server.https ? 'https' : 'http',
+        '://',
+        this.config.server.host,
```

Comment:
hostname need to be check that only included valid chars

Reply:
owners should know what it should be